### ISSUE TITLE
Cannot upgrade from 2.15.0 to 2.15.1 on PHP 7.2.10

### ISSUE THREAD

**Issue by SulphurDioxide** [mautic/mautic#7599]
## Bug Description

| Q   | A
| --- | ---
| Mautic version | 2.15.0
| PHP version | 7.2.10
| Browser | Chrome 75.0.3770.80

### Steps to reproduce
1. Install a fresh version of Mautic 2.15.0
2. Attempt to upgrade to Mautic 2.15.1 via GUI
 
### Log errors

> Symfony\\Component\\Debug\\Exception\\FatalThrowableError: Type error: Too few arguments to function Mautic\\LeadBundle\\Segment\\Decorator\\DateDecorator::__construct(), 2 passed in /pathToMautic/app/cache/prod/appProdProjectContainer.php on line 12988 and exactly 3 expected - in file /pathToMautic/app/bundles/LeadBundle/Segment/Decorator/DateDecorator.php - at line 37

### Mautic Log errors around the same time:
mautic.NOTICE: PHP Notice - Undefined index: objectName - in file /<pathToMautic>/app/bundles/DashboardBundle/Views/Dashboard/recentactivity.html.php - at line 44 [] []

### What I've tried
I initially tried doing the update via the GUI but the automated process hung at 'putting the files in place' and was producing server 500 errors in the JS console.

I followed every step on the [uh-oh blog post](https://johnlinhart.com/blog/uh-oh-mautic-upgrade-was-not-successful) with SSH access. None of the steps seemed to change the behaviour of Mautic (showing the generic error screen).

I was surprised there wasn't more logging in the mautic log leading up to the update but the only error there is the one I pasted above.  I reverted back to the backup of the system and manually copied the files over. As soon as the files were in place, the same issue started again before the database had been updated.

### Final notes
 - I saw that within the 'upgrade.php' file the `MAUTIC_MAXIMUM_PHP` is set to `7.1.999` which is lower than my current version of PHP although I didn't see the error this outputs in any of the logs or raw output.
 - I also noted that the `MAUTIC_MAXIMUM_PHP` version is increase to `7.2.999` in #7121 
 - The database user has all permissions for this database (apart from grant).

**Comment by andremacola** [comment]
you have to download the new upgrade.php

**Comment by SulphurDioxide** [comment]
@andremacola - Okay, I'll give that a try. Could you explain why this will fix the issue, please?

**Comment by SulphurDioxide** [comment]
Unfortunately, this didn't fix my issue. I rolled back to a previous version. Since the release of 2.15.2 i figured I'd try again. This time, the whole process seemed to work - at least, the command line tool reported that version 2.15.2 had been installed correctly.

When I went to visit my domain, however, it was stuck in a redirect loop. There were no errors here. I'm running Mautic behind a reverse proxy and I had a feeling this was to blame somehow. 

After switching protocol of site_url to http instead of https and clearing the cache, it worked again but now the various script generators were using 'http' links in the code snippets they output. To fix the issue I added `$_SERVER['HTTPS'] = 'on'` into the index.php file and this worked. 

I've noted in a few different issues people suggest adding `0.0.0.0/0` to `trusted_proxies` but this didn't make any difference to my server and the problem remained until I added the `$_SERVER` line.
